From 848fa10beb5915e9277485bd1f34961ff83dc420 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 28 Dec 2015 08:56:27 +0000
Subject: [PATCH] Assume a release value of '0' when unset

Some deb-based distros allow the release field to be optional.
---
 libappstream-builder/asb-package-deb.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/libappstream-builder/asb-package-deb.c b/libappstream-builder/asb-package-deb.c
index d854788..32c142a 100644
--- a/libappstream-builder/asb-package-deb.c
+++ b/libappstream-builder/asb-package-deb.c
@@ -87,7 +87,14 @@ asb_package_deb_ensure_simple (AsbPackage *pkg, GError **error)
 				asb_package_set_epoch (pkg, j);
 				asb_package_set_version (pkg, tmp + 1);
 			}
-			asb_package_set_release (pkg, vr[1]);
+			if (vr[1] != NULL) {
+				asb_package_set_release (pkg, vr[1]);
+			} else {
+				/* packages don't actually have to have a
+				 * release value like rpm; in this case fake
+				 * something plausible */
+				asb_package_set_release (pkg, "0");
+			}
 			continue;
 		}
 		if (g_str_has_prefix (lines[i], "Depends: ")) {
